### ISSUE TITLE
Unable to open splunk localhost:8000 on MacBook Pro M1 safari browser

### ISSUE THREAD

**Issue by shravankumar147** [splunk/splunk-connect-for-ethereum#86]
I have installed Docker, and followed instructions from https://github.com/splunk/splunk-connect-for-ethereum/tree/master/examples/docker-compose-basic

When I tried to open  http://localhost:8000, Safari(Mac) is not connecting. 

Any help is greatly appreciated.

**Comment by atoulme** [comment]
Hello @shravankumar147, did you follow step 2. and do you see the Splunk container showing as started?
What is the output of `docker ps`?

**Comment by shravankumar147** [comment]
Hi @atoulme,

Yes I did follow the step 2,


>> docker ps            
CONTAINER ID   IMAGE                                COMMAND                  CREATED       STATUS                            PORTS                                                                                                                                    NAMES
65adbba052c6   ghcr.io/splunkdlt/ethlogger:latest   "ethlogger"              2 hours ago   Up About a minute (healthy)                                                                                                                                                ethlogger
ad46ed9e23d6   splunk/splunk:latest                 "/sbin/entrypoint.sh…"   2 hours ago   Up 4 seconds (health: starting)   0.0.0.0:8000->8000/tcp, :::8000->8000/tcp, 8065/tcp, 8089/tcp, 8191/tcp, 9887/tcp, 0.0.0.0:8088->8088/tcp, :::8088->8088/tcp, 9997/tcp   splunk

**Comment by atoulme** [comment]
You can see the Splunk container is currently showing as `starting`. Therefore port 8000 is not yet used.
You can inspect the Splunk container logs with `docker logs splunk`.

**Comment by shravankumar147** [comment]
On the other hand, I looked at docker logs,

2021-08-04T18:59:27.683Z ethlogger:platforms:generic:warn Ethereum node failed to respond to protocol version request JsonRpcError: The method admin_nodeInfo does not exist/is not available

at Object.checkError (/ethlogger/lib/eth/jsonrpc.js:31:15)

at HttpTransport.sendInternal (/ethlogger/lib/eth/http.js:109:27)

at processTicksAndRejections (internal/process/task_queues.js:93:5)

at async HttpTransport.sendBatch (/ethlogger/lib/eth/http.js:62:24)

at async executeBatchRequest (/ethlogger/lib/eth/client.js:17:25)

at async Timeout.BatchedEthereumClient.processQueue (/ethlogger/lib/eth/client.js:97:20) {

code: -32601,

data: undefined

}


2021-08-04T18:59:27.706Z ethlogger:introspect:warn Failed to initialize node platform adapter: The method admin_nodeInfo does not exist/is not available


2021-08-04T18:59:27.707Z ethlogger:introspect:info Attempting to use generic node platform adapter: generic:unknown


2021-08-04T18:59:28.339Z ethlogger:platforms:generic:warn RPC method eth_pendingTransactions is not supported by ethereum node: The method eth_pendingTransactions does not exist/is not available


2021-08-04T18:59:28.793Z ethlogger:cli:info Detected node platform='generic:Geth' protocol=65 chainId=1 networkId=1 chain=mainnet network=eth


2021-08-04T18:59:28.841Z ethlogger:hec:warn HEC service not (yet) available: HEC service https://splunk:8088/services/collector/event/1.0 not available: FetchError: request to https://splunk:8088/services/collector/health failed, reason: getaddrinfo ENOTFOUND splunk


2021-08-04T18:59:28.842Z ethlogger:hec:info Waiting for HEC service https://splunk:8088/services/collector/event/1.0 to become available (timeout 120000 ms)

Was there anything I am missing.

**Comment by atoulme** [comment]
This line in the Splunk logs indicate the Splunk server failed to start:
`splunkd 7415 was not running`
You can try to restart the splunk container with `docker restart splunk`.

I see you changed the title of the issue to indicate that your laptop is an M1 laptop, using an Arm processor. This might be the problem, as Splunk might not run yet on Arm architectures. Please try on a different machine using the x86 architecture.

**Comment by shravankumar147** [comment]
It is sad to know that, my windows could not get docker installed, Initially Mac failed to run, but someone on stakoverflow suggested to use,

platform: linux/x86_64

It seemed to be working, but on the background the errors are stopping me. 

What else I can do to test this out, any suggestions. 

Thanks for your help.

Best
Shravan

**Comment by atoulme** [comment]
I believe you are asking for guidance beyond our example and about how to manage best a containerized environment with native processes relying on a x86 architecture.
Off the top of my head, I would offer that you host a virtual machine, using Fusion or VirtualBox, on your Windows machine where you can easily install a Linux environment, install Docker and run the sample. You can also start a virtual machine using a cloud provider to install Docker and run this example.

Best of luck! Please let us know how it turns out.

**Comment by shravankumar147** [comment]
Hi @atoulme ,

Thanks for your valuable recommendations, Here is what I did to get it up and running. 

- I have switched back to my window machine
- Installed Docker Desktop 2.4.0 version, as I struggled with latest version installation.
- used this : https://github.com/splunk/splunk-connect-for-ethereum.git
- Made a few changes with infura project id in docker-compose yaml file
- Followed instructions from here: https://github.com/splunk/splunk-connect-for-ethereum/tree/master/examples/docker-compose-basic
- This time it worked like charm

Appreciate your quick responses, helped me to make another choice to get things done. 

Best
Shravan
